python/src/iceberg/types.py

```diff
@@ -0,0 +1,824 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from decimal import Decimal
+from enum import Enum, unique
+import uuid
+
+
+@unique
+class TypeID(Enum):
+    BOOLEAN = {"java_class": "Boolean.class", "python_class": bool, "id": 1}
+    INTEGER = {"java_class": "Integer.class", "python_class": int, "id": 2}
+    LONG = {"java_class": "Long.class", "python_class": int, "id": 3}
+    FLOAT = {"java_class": "Float.class", "python_class": float, "id": 4}
+    DOUBLE = {"java_class": "Double.class", "python_class": float, "id": 5}
+    DATE = {"java_class": "Integer.class", "python_class": int, "id": 6}
+    TIME = {"java_class": "Long.class", "python_class": int, "id": 7}
+    TIMESTAMP = {"java_class": "Long.class", "python_class": int, "id": 8}
+    STRING = {"java_class": "CharSequence.class", "python_class": str, "id": 9}
+    UUID = {"java_class": "java.util.UUID.class", "python_class": uuid.UUID, "id": 10}
+    FIXED = {"java_class": "ByteBuffer.class", "python_class": bytes, "id": 11}
+    BINARY = {"java_class": "ByteBuffer.class", "python_class": bytearray, "id": 12}
+    DECIMAL = {"java_class": "BigDecimal.class", "python_class": Decimal, "id": 13}
+    STRUCT = {"java_class": "Void.class", "python_class": None, "id": 14}
+    LIST = {"java_class": "Void.class", "python_class": None, "id": 15}
+    MAP = {"java_class": "Void.class", "python_class": None, "id": 16}
+
+
+class Type(object):
+    length: int
+    scale: int
+    precision: int
+
+    def __init__(self):
+        pass
+
+    def type_id(self):
+        pass
+
+    def is_primitive_type(self):
+        return False
+
+    def as_primitive_type(self):
+        raise ValueError("Not a primitive type: " + self)
```

Comment:
I think these `as_` and `is_` methods are valuable if we have type annotations. Is the plan to add them later?